src/util.rs

```diff
@@ -345,6 +345,52 @@ pub unsafe fn get_gc_userdata<T: Any>(state: *mut ffi::lua_State, index: c_int)
     ud
 }
 
+pub unsafe fn init_userdata_metatable_index(
+    state: *mut ffi::lua_State
+) -> Result<()> {
+    match ffi::luaL_dostring(state, cstr!(r#"return function (__index, field_getters, methods)
```

Comment:
`luaL_dostring` is marked as `m` and can trigger memory errors (if it's set via `set_memory_limit()`) which cause unprotected crash.